InternTrack.Core.Api/Services/Foundations/Interns/InternService.cs

```diff
@@ -52,5 +52,19 @@ public partial class InternService : IInternService
 
         public IQueryable<Intern> RetrieveAllInternsAsync() =>
             TryCatch(() => this.storageBroker.SelectAllInternsAsync());
+
+        public ValueTask<Intern> ModifyInternAsync(Intern intern) =>
+        TryCatch(async () =>
+        {
+            ValidateInternOnModify(intern);
+
+            Intern maybeIntern =
+                await this.storageBroker.SelectInternByIdAsync(intern.Id);
+
```

Comment:
Move ValidateStorageIntern here, it is to understand this function at this level than grouping it into another function.